lib/features/steps/runner_steps.rb

```diff
@@ -62,3 +62,65 @@
   $logger.warn 'Sleep was used! Please avoid using sleep in tests!'
   sleep(seconds)
 end
+
+# Starts an interactive terminal
+When('I start a new terminal') do
+  Runner.stop_interactive_session
+  Runner.get_interactive_session
+end
+
+# Stops currently running interactive terminal
+When('I stop the current terminal') do
+  Runner.stop_interactive_session
+end
+
+# Run a command on the terminal
+#
+# @step_input command [String] The command to run on the terminal
+When('I input {string} interactively') do |command|
+  current_terminal = Runner.get_interactive_session
+  success = current_terminal.run_command(command)
+  assert(success, 'The terminal had already closed')
+end
+
+# Get the current buffer values in the terminal
+#
+# @step_input expected_chars [String] The chars present in current buffer
+Then('the terminal is outputting {string}') do |expected_chars|
+  current_terminal = Runner.get_interactive_session
+  assert_equal(expected_chars, current_terminal.current_buffer)
+end
+
+# Verify a string appears in the stdout logs
+#
+# @step_input expected_line [String] The string present in stdout logs
+Then('the terminal has output {string}') do |expected_line|
+  current_terminal = Runner.get_interactive_session
+  match = current_terminal.parsed_output.any? { |line| line == expected_line }
+  assert(match, "No output lines matched #{expected_line}")
```

Comment:
I think we should always output the actual lines received as well to make debugging possible.  It might mean a lot of output, but as long as it's formatted in a friendly manner (e.g. multiline and clear delimiters at start/end) I would rather have lots of output than none at all.

Reply:
(Happy for this to be handled in a separate PR, of course).

Reply:
This should be being output via the CLI instance, whenever it receives a newline char it will store the current line, `$logger.debug` the line (against the shell PID), and then clear for the next line